src/lib.rs

```diff
@@ -86,9 +87,12 @@ impl Pattern {
     /// // sequence when it's followed by non-matching characters:
     /// assert!(pattern.display_matches(&"hello world! aaaaab"));
     /// ```
-    pub fn new(pattern: &str) -> Result<Self, Error> {
-        let automaton = DenseDFA::new(pattern)?;
-        Ok(Pattern { automaton })
+    pub fn new(pattern: &str) -> Result<Self, dense::BuildError> {
+        let automaton = dense::DFA::new(pattern)?;
+        let start = automaton
+            .universal_start_state(Anchored::No)
+            .expect("I hope this works");
```

Comment:
I think we should probably return our own error type, which would be a `#[non-exhaustive]` enum of either a `dense::BuildError` or our own error variant indicating that the pattern is invalid because the automaton would have no universal start state. That way, we can remove the `expect` here.